src/monet.js

```diff
@@ -177,6 +177,22 @@
         return listFindC(l, fn).run()
     }
 
+    function listContainsC(l, val) {
+        if (l.isNil) {
+            return Return(false)
+        }
+        var h = l.head()
+        return h === val ?
```

Comment:
`Immutable` has quite sophisticated implementation of `is`-> https://github.com/facebook/immutable-js/blob/master/src/is.js

but I think we don't need that much, I think that sth like this would be enough:

```javascript
  if (a === b || (a !== a && b !== b)) {
    return true;
  }
  if (!a || !b) {
    return false;
  }
  if (isFunction(a.equals) && isFunction(b.equals)) {
    return a.equals(b);
  }
  return false;
```

The `a !== a && b !== b` is about `NaN`.